docs/auto-tag-products-marked-continue-selling-when-out-of-stock/script.liquid

```diff
@@ -0,0 +1,192 @@
+{% assign tag_for_continue_selling_products = options.tag_for_continue_selling_products__required %}
+
+{% comment %}
+  -- for product create/update, filter the products query with the product ID that caused the event
+{% endcomment %}
+
+{% if event.topic contains "shopify/products/" %}
+  {% assign search_query = product.id | prepend: "id:" %}
+{% endif %}
+
+{% comment %}
+  -- query product(s) in the shop; variants will be queried separately as needed to support up to 2K variants
+{% endcomment %}
+
+{% assign cursor = nil %}
+{% assign products = array %}
+
+{% for n in (1..100) %}
+  {% capture query %}
+    query {
+      products(
+        first: 250
+        after: {{ cursor | json }}
+        query: {{ search_query | json }}
+      ) {
+        pageInfo {
+          hasNextPage
+          endCursor
+        }
+        nodes {
+          id
+          tags
+        }
+      }
+    }
+  {% endcapture %}
+
+  {% assign result = query | shopify %}
+
+  {% if event.preview %}
+    {% capture result_json %}
+      {
+        "data": {
+          "products": {
+            "nodes": [
+              {
+                "id": "gid://shopify/Product/1234567890"
+              }
+            ]
+          }
+        }
+      }
+    {% endcapture %}
+
+    {% assign result = result_json | parse_json %}
+  {% endif %}
+
+  {% assign products = products | concat: result.data.products.nodes %}
+
+  {% if result.data.products.pageInfo.hasNextPage %}
+    {% assign cursor = result.data.products.pageInfo.endCursor %}
+  {% else %}
+    {% break %}
+  {% endif %}
+{% endfor %}
+
+{% comment %}
+  -- process each product by querying as many variants as needed to determine whether the product should be tagged
+{% endcomment %}
+
+{% for product in products %}
+  {% assign product_set_to_continue_selling = nil %}
+  {% assign cursor = nil %}
+
+  {% for n in (1..8) %}
```

Comment:
FYI only, no change needed... the original task this new one is based on was developed before Shopify increased the variant query limit on a single product object to 2048. This inner paginated query could be reduced to a single query for all variants. I checked the `requestedQueryCost` and it only went up to 18 from 14 when increasing the 250 to 2048.